Source/WTF/wtf/URLParser.cpp

```diff
@@ -2883,21 +2883,28 @@ auto URLParser::parseURLEncodedForm(StringView input) -> URLEncodedForm
 {
     URLEncodedForm output;
     for (StringView bytes : input.split('&')) {
-        auto equalIndex = bytes.find('=');
-        if (equalIndex == notFound) {
-            auto name = formURLDecode(makeStringByReplacingAll(bytes, '+', 0x20));
-            if (name)
-                output.append({ name.value(), emptyString() });
-        } else {
-            auto name = formURLDecode(makeStringByReplacingAll(bytes.left(equalIndex), '+', 0x20));
-            auto value = formURLDecode(makeStringByReplacingAll(bytes.substring(equalIndex + 1), '+', 0x20));
-            if (name && value)
-                output.append({ name.value(), value.value() });
-        }
+        if (auto nameAndValue = parseQueryNameAndValue(bytes))
+            output.append(WTFMove(*nameAndValue));
     }
     return output;
 }
 
+std::optional<KeyValuePair<String, String>> URLParser::parseQueryNameAndValue(StringView bytes)
+{
+    auto equalIndex = bytes.find('=');
+    if (equalIndex == notFound) {
+        auto name = formURLDecode(makeStringByReplacingAll(bytes, '+', 0x20));
+        if (name)
+            return { { name.value(), emptyString() } };
+    } else {
+        auto name = formURLDecode(makeStringByReplacingAll(bytes.left(equalIndex), '+', 0x20));
+        auto value = formURLDecode(makeStringByReplacingAll(bytes.substring(equalIndex + 1), '+', 0x20));
+        if (name && value)
+            return { { name.value(), value.value() } };
```

Comment:
`return { { WTFMove(*name), WTFMove(*value) } };`

Would be a tad more efficient.

Reply:
Fixed!